File: backend/pm-risk/src/main/java/com/pm/risk/domain/Risk.java
package com.pm.risk.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.pm.common.base.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("risk")
@Schema(description = "风险")
public class Risk extends BaseEntity {

    @Schema(description = "所属项目ID")
    private Long projectId;

    @Schema(description = "风险标题")
    private String title;

    @Schema(description = "风险描述")
    private String description;

    @Schema(description = "类别（technical/resource/schedule/quality/external）")
    private String category;

    @Schema(description = "发生概率（high/medium/low）")
    private String probability;

    @Schema(description = "影响程度（high/medium/low）")
    private String impact;

    @Schema(description = "风险等级（critical/high/medium/low）")
    private String riskLevel;

    @Schema(description = "状态（open/monitoring/mitigated/closed/realized）")
    private String status;

    @Schema(description = "风险负责人ID")
    private Long ownerId;

    @Schema(description = "应对措施")
    private String mitigationPlan;

    @Schema(description = "应急预案")
    private String contingencyPlan;

    @Schema(description = "识别日期")
    private LocalDate identifiedDate;

    @Schema(description = "复审日期")
    private LocalDate reviewDate;

    @Schema(description = "关闭日期")
    private LocalDate closedDate;
}

File: frontend/src/types/risk.ts
/** 风险状态 */
export type RiskStatus = 'open' | 'monitoring' | 'mitigated' | 'closed' | 'realized'

/** 风险等级 */
export type RiskLevel = 'critical' | 'high' | 'medium' | 'low'

/** 风险类别 */
export type RiskCategory = 'technical' | 'resource' | 'schedule' | 'quality' | 'external'

/** 概率/影响级别 */
export type ProbabilityImpact = 'high' | 'medium' | 'low'

/** 问题状态 */
export type IssueStatus = 'open' | 'in_progress' | 'resolved' | 'closed' | 'reopen'

/** 问题严重程度 */
export type IssueSeverity = 'critical' | 'major' | 'minor' | 'trivial'

/** 问题类别 */
export type IssueCategory = 'technical' | 'process' | 'resource' | 'communication' | 'other'

/** 风险 */
export interface Risk {
  id: number
  projectId: number
  title: string
  description?: string
  category?: RiskCategory
  probability: ProbabilityImpact
  impact: ProbabilityImpact
  riskLevel?: RiskLevel
  status: RiskStatus
  ownerId?: number
  mitigationPlan?: string
  contingencyPlan?: string
  identifiedDate?: string
  reviewDate?: string
  closedDate?: string
  createdAt: string
}

/** 风险创建请求 */
export interface RiskCreateRequest {
  title: string
  description?: string
  category?: RiskCategory
  probability?: ProbabilityImpact
  impact?: ProbabilityImpact
  ownerId?: number
  mitigationPlan?: string
  contingencyPlan?: string
  identifiedDate?: string
  reviewDate?: string
}

/** 风险状态更新请求 */
export interface RiskStatusUpdateRequest {
  status: RiskStatus
}

/** 问题 */
export interface Issue {
  id: number
  projectId: number
  title: string
  description?: string
  category?: IssueCategory
  severity: IssueSeverity
  status: IssueStatus
  assignedTo?: number
  reportedBy?: number
  resolution?: string
  resolvedAt?: string
  closedAt?: string
  dueDate?: string
  createdAt: string
}

/** 问题创建请求 */
export interface IssueCreateRequest {
  title: string
  description?: string
  category?: IssueCategory
  severity?: IssueSeverity
  assignedTo?: number
  dueDate?: string
}

/** 问题状态更新请求 */
export interface IssueStatusUpdateRequest {
  status: IssueStatus
  resolution?: string
}
